pkg/datastore/datastore.go

```diff
@@ -79,22 +79,31 @@ func (d *Conn) Exists(ctx context.Context, collection string, id int) (bool, err
 	return respData.Exists, nil
 }
 
-// Create sends a create event to the datastore.
-func (d *Conn) Create(ctx context.Context, fqid string, fields map[string]json.RawMessage) error {
-	f, err := json.Marshal(fields)
-	if err != nil {
-		return fmt.Errorf("marshalling fields: %w", err)
+// Create sends multiple create events to the datastore.
+func (d *Conn) Create(ctx context.Context, creatables map[string]map[string]json.RawMessage, migrationIndex int) error {
+	events := ""
+	for fqid, fields := range creatables {
+		decodedFields, err := json.Marshal(fields)
+		if err != nil {
+			return fmt.Errorf("marshalling fields: %w", err)
+		}
+		events += fmt.Sprintf(
+			`{"type":"create","fqid":"%s","fields":%s},`,
+			fqid, decodedFields,
+		)
+
 	}
+	events = strings.TrimSuffix(events, ",")
```

Comment:
Proposal:

I know you sad I coded that. But when I look it at now, I would save the "events" in a slice and join them later.

```go
var events []string
for fqid, fields := range creatables {
 ....
  event := fmt.Sprintf(`{"type"...`)
  events = append(events, event
}

bodyEvent := strings.Join(events, ",")
```

Or:

```go
reqBody := struct{
  UserID int `json:"user_id"`
  Information map[string]string `json:"information"`
  LockedFields map[string]string `json:"locked_fields"`
  Events []json.RawMessage `json:"events""`
  MigrationIndex int `json:"migration_index"`
} {
  0,
  map[string]string{},
  map[string]string{},
  events,
  migrationIndex,
}
bs, err := json.Marshal(reqBody)
if err != nil {...}
...
```

This is a bit more expose, but maybe it is clearer. For example, if one event would contain a "]" (I know, this is not possible), then your "json-template" would fail. At this place, this is not possible, but I did something similar in OS3 with the username and it crashed.

Reply:
See next PR.